### PR TITLE
i2 add unit tests for all functions

### LINKED ISSUE #2: Add unit tests for all functions on IOperationResult<T>
All `BindAsync` overloads must be covered just like `Bind`

### PR DESCRIPTION
**Pull request by TheSonOfDeimos**
Related #2

### CODE REVIEW

**Review by vkamiansky — CHANGES_REQUESTED** [review]
There are 3 `BindAsync` methods. This one is covered. We need the remaining two covered as well.

**Comment by TheSonOfDeimos** [comment]
It's not complete yet. That's why pull request with a WIP tag

File: src/Fls.Results.Test/ResultFunctionsTests.cs [hunk 1]
```diff
@@ -53,5 +54,122 @@ public void BindTest()
 
             Assert.Equal(expectedResult, actualResult);
         }
+
+        [Fact]
+        public async void BindAsyncTest_1()
```
Comment (vkamiansky) [inline src/Fls.Results.Test/ResultFunctionsTests.cs]: Suppose one of these tests fails and the output says 'BindAsyncTest_1 failed' how do I make out what scenario isn't working. Make test names say in their names what they mean to test and don't be afraid of long names, just of the meaningless ones.

File: src/Fls.Results.Test/ResultFunctionsTests.cs [hunk 2]
```diff
@@ -53,5 +54,122 @@ public void BindTest()
...
+                    It.Is<Func<string, Task<IOperationResult<int>>>>(y => ( y(testError).Result as OperationResult.ErrorResult<int>).Message == testError),
+                    It.Is<Func<Exception, Task<IOperationResult<int>>>>(y => (y(testException).Result as OperationResult.FailureResult<int>).Exception == testException)
+                ), Times.Once);
+
+            
+            Assert.Equal(expectedResult, actualResult);
+        }
+
+        [Fact]
+        public async void BindAsyncTest_2()
```
Comment (vkamiansky) [inline src/Fls.Results.Test/ResultFunctionsTests.cs]: Name change suggested.

File: src/Fls.Results.Test/ResultFunctionsTests.cs [hunk 3]
```diff
@@ -53,5 +54,122 @@ public void BindTest()
...
+                    It.Is<Func<string, Task<IOperationResult<int>>>>(y => ( y(testError).Result as OperationResult.ErrorResult<int>).Message == testError),
+                    It.Is<Func<Exception, Task<IOperationResult<int>>>>(y => (y(testException).Result as OperationResult.FailureResult<int>).Exception == testException)
+                ), Times.Once);
+
+            
+            Assert.Equal(expectedResult, actualResult);
+        }
+        
+        [Fact]
+        public async void BindAsyncTest_3()
```
Comment (vkamiansky) [inline src/Fls.Results.Test/ResultFunctionsTests.cs]: Name change suggested.

**Review by vkamiansky — CHANGES_REQUESTED** [review]
We might want our method names more telling.

**Review by vkamiansky — APPROVED** [review]

**Pull request merged**